props/nested-components.md

```diff
@@ -0,0 +1,195 @@
+
+<!-- prettier-ignore-start -->
+```js
+function App() {
+  return (
+    <main>
+      <h1>Attendance</h1>
+      <ClassInfo></ClassInfo>
+      <StudentList></StudentList>
+    </main>
+  );
+}
+```
+<!-- prettier-ignore-end -->
+
+When an `App` component is rendered, it begins with a `<main>` element.
+
+The `App` component has two _nested_ components, or components rendered inside it:
+
+1. One instance of a `ClassInfo` component
+1. One instance of a `StudentList` component
+
+The `ClassInfo` and `StudentList` components are _siblings_ to each other.
+
+As React webapps increase in complexity, more and more components are defined and rendered.
+
+Deciding how to design a React webapp and its components can be challenging. We can learn about and apply two design patterns to help us:
+
+1. Presentational components
+1. Container components
+
+These are design patterns; these labels don't affect the syntax used to define them. The difference between these two kinds of components is the nature of their responsibilities, and how we use them.
+
+## Presentational Components
+
+Presentational components are components that are responsible for displaying data.
+
+Presentational components are not responsible for modifying data.
+
+## Container Components
+
+Container components are components responsible for:
+
+- Using presentation components
+- Holding data
+- Modifying and managing data
+- Sending data to presentation components
+
+Container components often have presentation components _nested_ inside them.
+
+## Example: `StudentList` and `Student` Components
+
+Sofia is a teacher creating an attendance app. Her app should display a list of students. At this moment, she starts with this implementation of a `StudentList` component in `src/components/StudentList.js`.
+
+(Class names have been omitted from this code snippet for clarity)
+
+<!-- prettier-ignore-start -->
+```js
+import React from 'react';
+import './StudentList.css';
+
+const StudentList = () => {
+    return (
+        <section>
+            <h2>Student List</h2>
+            <ul>
+                <li>Student A</li>
+                <li>Student B</li>
+                <li>Student C</li>
+            </ul>
+        </section>
+    )
+}
+
+export default StudentList;
+```
+<!-- prettier-ignore-end -->
+
+This component displays a list of students, where each student is in an `<li>` element.
+
+However, Sofia knows that each student list item should display a lot of data, such as their name and email address.
+
+Sofia can apply the container and presentational component design patterns, and:
+
+1. Create a `Student` component, which will act as a presentational component
+1. Use the `Student` component in `StudentList`, which will act as a container component
+
+### `Student` Component
+
+Sofia's `Student` component should display a student's name and email address. She'll create a new file, `src/components/Student.js`, and write this code:
```

Comment:
switch to a present tense narration so we feel like we watch her doing these steps.
```suggestion
Sofia's `Student` component should display a student's name and email address. She creates a new file, `src/components/Student.js`, and writes this code:
```